agents/statsd/src/statsd_agent.h

```diff
@@ -34,8 +36,13 @@ static const nlohmann::json default_agent_config = {
   { "pauseMetrics", false }
 };
 
-class StatsDTcp {
+using WeakStatsDTcp = std::weak_ptr<StatsDTcp>;
+
+class StatsDTcp : public std::enable_shared_from_this<StatsDTcp> {
```

Comment:
IIUC the [documentation](https://en.cppreference.com/w/cpp/memory/enable_shared_from_this), If you're inheriting from `std::enable_shared_from_this` the `StatsDTcp` instance should be stored in a `shared_ptr`.

Reply:
Looking over it more carefully, I realize now that it's not necessary to inherit from `std::enable_shared_from_this` if the life of the timer is controlled more carefully. Going to do the rewrite.